src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs

```diff
@@ -239,6 +260,55 @@ public sealed class ShardOptions
         /// graceful shutdown of a <see cref="Sharding.ShardRegion"/>, e.g. <see cref="PoisonPill"/>.
         /// </summary>
         public object? HandOffStopMessage { get; set; }
+
+        /// <summary>
+        /// Throw an exception if the internal state machine in the Shard actor does an invalid state transition.
+        /// Mostly for the Akka test suite, if off the invalid transition is logged as a warning instead of throwing and
+        /// crashing the shard.
+        /// </summary>
+        public bool? FailOnInvalidEntityStateTransition { get; set; }
+        
+        internal void Apply(AkkaConfigurationBuilder builder)
+        {
+            var sb = new StringBuilder();
+
+            if (Role is { })
+                sb.AppendLine($"role = {Role.ToHocon()}");
+            
+            if(RememberEntities is { })
+                sb.AppendLine($"remember-entities = {RememberEntities.ToHocon()}");
+            
+            if(RememberEntitiesStore is { })
```

Comment:
So if the value is not populated, don't set it in HOCON - that looks correct to me for backwards compat IMHO